test/extensions/quic_listeners/quiche/active_quic_listener_test.cc

```diff
@@ -59,17 +78,48 @@ class ActiveQuicListenerTest : public testing::TestWithParam<Network::Address::I
         local_address_(Network::Test::getCanonicalLoopbackAddress(version_)),
         connection_handler_(*dispatcher_) {}
 
+  template <typename A, typename B>
+  std::unique_ptr<A> static_unique_pointer_cast(std::unique_ptr<B>&& source) {
+    return std::unique_ptr<A>{static_cast<A*>(source.release())};
+  }
+
   void SetUp() override {
+    envoy::config::bootstrap::v3::LayeredRuntime config;
+    config.add_layers()->mutable_admin_layer();
+    loader_ = std::make_unique<Runtime::ScopedLoaderSingleton>(Runtime::LoaderPtr{
+        new Runtime::LoaderImpl(*dispatcher_, tls_, config, local_info_, init_manager_, store_,
+                                generator_, validation_visitor_, *api_)});
     listen_socket_ =
         std::make_shared<Network::UdpListenSocket>(local_address_, nullptr, /*bind*/ true);
     listen_socket_->addOptions(Network::SocketOptionFactory::buildIpPacketInfoOptions());
     listen_socket_->addOptions(Network::SocketOptionFactory::buildRxQueueOverFlowOptions());
 
-    quic_listener_ = std::make_unique<ActiveQuicListener>(
-        *dispatcher_, connection_handler_, listen_socket_, listener_config_, quic_config_, nullptr);
+    EXPECT_CALL(listener_config_, listenSocketFactory())
```

Comment:
this is needed to wire up mocks, that are used inside listener factory when creating quic listener

Reply:
For this type of thing it's better to use ON_CALL(...).WillByDefault()